ACPs/236-continuous-staking/README.md

```diff
@@ -0,0 +1,173 @@
+
+## Abstract
+
+This proposal introduces continuous staking for validators on the Avalanche P-Chain. Validators can stake their tokens
+continuously, allowing their stake to compound over time, accruing rewards once per specified cycle.
+
+## Motivation
+
+The current staking system on the Avalanche P-Chain restricts flexibility for stakers, limiting their ability to respond
+to changing market conditions or liquidity needs. Managing a large number of nodes is also challenging, as re-staking at
+the end of each period is labor-intensive, time-consuming, and poses security risks due to
+the required transaction signing. Additionally, tokens can remain idle at the end of a staking period
+until stakers initiate the necessary transactions to stake them again.
+
+## Specification
+
+Continuous staking introduces a mechanism that allows validators to remain staked indefinitely, without having to
+manually submit new staking transactions at the end of each period.
+
+Instead of committing to a fixed endtime upfront, validators specify a cycle duration (period) when they submit an
+`AddContinuousValidatorTx`. At the end of each cycle, the validator is automatically restaked for a new cycle of the
+same duration, unless the validator has submitted a `StopContinuousValidatorTx`. If a validator submits a
+`StopContinuousValidatorTx` during a cycle, the validator will continue validating until the end of the current cycle,
+at which point the validator exits and the funds are unlocked. The minimum and maximum cycle lengths follow the same
+protocol parameters as before (`MinStakeDuration` and `MaxStakeDuration`).
+
+Delegator interaction remains unchanged, and the same constraints apply: a delegation period must fit entirely within
+the validator’s cycle. Delegators cannot delegate across multiple cycles, since there is no guarantee that a validator
+will continue validating after the current cycle. Essentially, it is not possible to delegate continuously.
+
+Rewards accrue once per cycle, and are automatically added to the validator's existing stake in subsequent cycles, both
+for validation rewards and for delegatee rewards. If the updated stake weight (previous stake + staking rewards +
+delegatee rewards) exceeds the maximum stake limit defined in the network configuration, the excess amount is
+automatically withdrawn and sent to `ValidatorRewardsOwner` and `DelegatorRewardsOwner`.
+
+Because of the way `RewardValidatorTx` is structured, multiple instances cannot be issued without resulting in identical
+transaction IDs. To resolve this, a new transaction type has been introduced for both rewarding and stopping continuous
+validators: `RewardContinuousValidatorTx`. Along with the validator’s creation transaction ID, it also includes a
+timestamp. For simplicity and consistency, any stake exceeding the maximum limit is withdrawn from the validator, and
+the resulting UTXOs are tied to the `RewardContinuousValidatorTx` ID.
+
```

Comment:
I think we should specify how uptime requirements work for continuous validators here as well.